test/utils/IntegrationTest.sol

```diff
@@ -0,0 +1,415 @@
+    /// @param deploymentSalt The deployment salt for create2.
+    /// @param contribution The amount to initialize the market.
+    /// @param asBase Initialize the market with base token.
+    function deployTargets(
+        bytes32 deploymentId,
+        bytes32 deploymentSalt,
+        uint256 contribution,
+        bool asBase
+    ) internal {
+        // Alice is the default deployer.
+        vm.startPrank(alice);
+
+        // Deploy Hyperdrive target contracts.
+        factory.deployTarget(
+            deploymentId,
+            deployerCoordinator,
+            poolConfig,
+            new bytes(0),
+            FIXED_RATE,
+            FIXED_RATE,
+            0,
+            deploymentSalt
+        );
+        factory.deployTarget(
+            deploymentId,
+            deployerCoordinator,
+            poolConfig,
+            new bytes(0),
+            FIXED_RATE,
+            FIXED_RATE,
+            1,
+            deploymentSalt
+        );
+        factory.deployTarget(
+            deploymentId,
+            deployerCoordinator,
+            poolConfig,
+            new bytes(0),
+            FIXED_RATE,
+            FIXED_RATE,
+            2,
+            deploymentSalt
+        );
+        factory.deployTarget(
+            deploymentId,
+            deployerCoordinator,
+            poolConfig,
+            new bytes(0),
+            FIXED_RATE,
+            FIXED_RATE,
+            3,
+            deploymentSalt
+        );
+        factory.deployTarget(
+            deploymentId,
+            deployerCoordinator,
+            poolConfig,
+            new bytes(0),
+            FIXED_RATE,
+            FIXED_RATE,
+            4,
+            deploymentSalt
+        );
+
+        // Alice gives approval to the deployer coordinator to fund the market.
+        config.token.approve(deployerCoordinator, 100_000e18);
+
+        // We expect the deployAndInitialize to fail with an
+        // UnSupported token error if depositing as base is not supported.
+        // If the base token is ETH we expect a NotPayable error.
+        if (!config.enableBaseDeposits && asBase) {
+            vm.expectRevert(
+                isBaseETH
+                    ? IHyperdrive.NotPayable.selector
+                    : IHyperdrive.UnsupportedToken.selector
+            );
+        }
+
+        // Deploy and initialize the market. If the base token is ETH we pass the
+        // contribution through the call.
+        hyperdrive = factory.deployAndInitialize{
+            value: asBase && isBaseETH ? contribution : 0
+        }(
+            deploymentId,
+            deployerCoordinator,
+            poolConfig,
+            new bytes(0),
+            contribution,
+            FIXED_RATE,
+            FIXED_RATE,
+            IHyperdrive.Options({
+                asBase: asBase,
+                destination: alice,
+                extraData: new bytes(0)
+            }),
+            deploymentSalt
+        );
```

Comment:
In the current tests, we make assertions about the LP shares received by the LPs and the initial quantity of LP shares. I think it would still be good to do this.